iroh-net/src/lib.rs

```diff
@@ -1,7 +1,10 @@
 #![recursion_limit = "256"]
 
-pub mod client;
+pub mod defaults;
 pub mod hp;
+pub mod magic_endpoint;
```

Comment:
can we make this private, if we reexport the `MagicEndpoint`?

Reply:
There's also the `MagicEndpointBuilder` struct and `accept_conn` functions in there (and possible `get_alpn` and `get_peer_id`, see below). Not sure if they should all be at the top level?
I think I'd defer this to later, currently pretty much everything in `iroh-net` is public anyway - and once things stabilize a bit we might want to change this overall and actually define the public API surface for `iroh-net`?